### DISCUSSION TITLE
Find a list of library procedures

### DISCUSSION THREAD

**Discussion by pepa65** [odin-lang/Odin#827] [Q&A]
I appreciate the documentation might not be up, but was wondering what would be the way to find a list of procedures that are build in for example "core:strings" or "core:strconv".

**Comment by Tetralux** [comment]
In the root folder of the repository, there is a `core` folder - each folder within that is a package you can import with `import`;
`import "core:strings"` will import the `repo/core/strings` package, for example.

You can look in any of them for the source code for any of the core libararies.

**Reply by Tetralux** [reply]
If you specifically want a list of all procedures, one option is to grep for `:: proc`.
Another is to use `odin doc repo/core/strings`, for example.

**Reply by pepa65** [reply]
OK, being in the root of the source tree and then `odin doc core/strings` gives a lot of output, with some comments too!

**Reply by Tetralux** [reply]
Yeah, when I said `repo/core/strings`, I meant `/path/to/the/odin/repo/core/strings` 😄 
(i.e: I don't know the path you cloned the repo into on your computer.)

**Comment by gingerBill** [comment]
A good way to show a list of declarations is by using the `odin doc` command line tool.

```
odin doc path/to/core/strings
```

to do see more options, see `odin doc -help`
